src/SonarScanner.MSBuild.Shim/AdditionalFilesService.cs

```diff
@@ -108,24 +108,27 @@ private static AdditionalFiles PartitionAdditionalFiles(List<FileInfo> allFiles,
         return new(sources, tests);
     }
 
-    private static string[] GetTestExtensions(AnalysisProperties properties) =>
-        properties is null
-            ? []
-            : properties
-                .Where(x => SupportedTestLanguages.Contains(x.Id))
-                .SelectMany(x => x.Value.Split(Comma, StringSplitOptions.RemoveEmptyEntries))
-                .SelectMany(x => SupportedTestInfixes.Select(infix => $".{infix}{EnsureDot(x)}"))
-                .Distinct()
-                .ToArray();
-
-    private static string[] GetExtensions(AnalysisProperties properties) =>
-        properties is null
-            ? []
-            : SupportedLanguages
-                .Select(x => properties.Find(property => property.Id == x))
-                .Where(x => x is { Value: { } })
-                .SelectMany(x => x.Value.Split(Comma, StringSplitOptions.RemoveEmptyEntries).Select(EnsureDot))
-                .ToArray();
+    private static string[] GetExtensions(AnalysisConfig config) =>
+        GetProperties(config, SupportedLanguages);
+
+    private static string[] GetTestExtensions(AnalysisConfig config) =>
+        GetProperties(config, SupportedTestLanguages)
+            .SelectMany(x => SupportedTestInfixes.Select(infix => $".{infix}{x}"))
+            .Distinct()
+            .ToArray();
+
+    private static string[] GetProperties(AnalysisConfig config, List<string> ids) =>
+        ids
+            .Select(x => GetProperty(config, x))
+            .Where(x => x is { Value: { } })
+            .SelectMany(x => x.Value.Split(Comma, StringSplitOptions.RemoveEmptyEntries))
+            .Select(EnsureDot)
+            .ToArray();
+
+    // Local settings take priority over Server settings.
+    private static Property GetProperty(AnalysisConfig config, string id) =>
+        config.LocalSettings?.Find(x => x.Id == id)
+        ?? config.ServerSettings?.Find(x => x.Id == id);
```

Comment:
This is the meat and potatoes of the PR.
We want to take into account local settings as well, as they might be passed in the command line.

Reply:
Does localSettings take all of these sources into account?
* cmdLineProperties,
* globalFileProperties,
* scannerEnvProperties,

Reply:
Yep!